sleap/gui/commands.py

```diff
@@ -4052,6 +4066,11 @@ def get_products_of_instances(
             track=-1,  # Get all instances regardless of track.
             require_multiple_views=True,
         )
+        # Possible race condition
+        if selected_cam not in instances:
+            raise RuntimeError(
+                f"Selected camera {selected_cam} not in instances at frame {frame_idx}."
+            )
```

Comment:
While this will help prevent a KeyError if the instance has not yet been added from the `selected_cam` - this does not handle the case when an instance is already available for the `selected_cam`, but we double click another `PredictedInstance` in the same view.